src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -235,11 +236,13 @@ public async Task<ChainedHeaderBlock> BlockMinedAsync(Block block)
                         return null;
                     }
 
+                    // This might throw HeaderValidationFailedException but we don't wanna catch it because miner should never produce a block with invalid header.
                     chainedHeader = this.chainedHeaderTree.CreateChainedHeaderWithBlock(block);
```

Comment:
I'm not sure if this is correct

Reply:
We want to catch it? 
Why? 

The only way we can't accept out block is if we already accepted alternative block from the network. 
We should never disvalidate our block